source/custom/visa_service.custom.cpp

```diff
@@ -458,21 +457,23 @@ ::grpc::Status VisaService::UsbControlIn(::grpc::ServerContext* context, const U
     ViUInt16 w_value = request->w_value();
     ViUInt16 w_index = request->w_index();
     ViUInt16 w_length = request->w_length();
-    std::string buffer(return_count, '\0');
+    std::vector<ViByte> buffer(w_length);
```

Comment:
This means the buffer is allocated twice, once here and again in the response. I think we can optimize this just a bit:
```
std::string* buffer = response->mutable_buffer();
buffer->resize(w_length);
...
auto status = library_->UsbControlIn(... buffer->data(), &return_count);
...
buffer->resize(return_count);
```

Reply:
And you should be able to do the same for Read.

Reply:
Commit 2.

Nice to save the extra allocation and copy. Do we consider to apply it for all the methods? Even beyond Visa?

Reply:
I hadn't looked at the other ones. The custom code generally looks good to me. There are copies of course for cases where the structures are too different. But I did find in generated code a couple of possibilities for improvement. I created a "tech debt" issue #1011